### PR TITLE
[TEVA 1099] Add contact number

### PR DESCRIPTION
**Pull request by cpjmcquillan**
This PR adds a contact number input in the applying for the job section of the create a job listing journey.

## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1099

## Changes in this PR
- Add contact number input

### CODE REVIEW

File: app/components/jobseekers/school_group_overview_component.html.haml [hunk 1]
```diff
@@ -13,13 +13,19 @@
           %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" }
             = t('school_groups.website')
           %td.govuk-table__cell
-            = link_to(t('jobs.website_link_text', organisation_name: @vacancy.parent_organisation.name), @vacancy.parent_organisation.website.presence || @vacancy.parent_organisation.url, class: 'govuk-link wordwrap', target: '_blank')
+            = link_to(t('schools.website_link_text', organisation_name: @vacancy.parent_organisation.name), @vacancy.parent_organisation.website.presence || @vacancy.parent_organisation.url, class: 'govuk-link wordwrap', target: '_blank')
```
Comment (david-mears-dfe) [inline app/components/jobseekers/school_group_overview_component.html.haml:16]: Can we save space (and the number of loads?) in the component template, by setting `@trust = vacancy.parent_organisation` in the component initialize method?

File: app/components/jobseekers/school_group_overview_component.html.haml [hunk 2]
```diff
@@ -13,13 +13,19 @@
...
       - if @vacancy.contact_email.present?
         %tr.govuk-table__row
           %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" }
-            = t('school_groups.email')
```
Comment (david-mears-dfe) [inline app/components/jobseekers/school_group_overview_component.html.haml:20]: As we are not using the translation elsewhere, please delete it from en.yml.

File: app/components/jobseekers/school_group_overview_component.html.haml [hunk 3]
```diff
@@ -13,13 +13,19 @@
...
+            = t('jobs.contact_email')
           %td.govuk-table__cell
             = mail_to @vacancy.contact_email, @vacancy.contact_email, class: 'govuk-link link-wrap', subject: t('jobs.contact_email_subject', job: @vacancy.job_title), body: t('jobs.contact_email_body', url: job_url(@vacancy))
+      - if @vacancy.contact_number.present?
+        %tr.govuk-table__row
+          %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" }
+            = t('jobs.contact_number')
+          %td.govuk-table__cell
+            = link_to @vacancy.contact_number, "tel:#{@vacancy.contact_number}", class: 'govuk-link'
```
Comment (david-mears-dfe) [inline app/components/jobseekers/school_group_overview_component.html.haml:28]: Where does `tel:` come from? Are you rendering 'tel:01234567890' to the page, or is `tel:` a `link_to` helper method that formats the number?

File: app/components/jobseekers/school_group_overview_component.html.haml (hunk 1 above)
Comment (david-mears-dfe) [reply app/components/jobseekers/school_group_overview_component.html.haml:16]: Or even use an attribute reader if you like, as mentioned in TWD dev meetup!

File: app/components/jobseekers/school_overview_component.html.haml [hunk 4]
```diff
@@ -37,13 +37,20 @@
           %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" }
             = t('schools.website')
           %td.govuk-table__cell
-            = link_to(t('jobs.website_link_text', organisation_name: @vacancy.parent_organisation.name), @vacancy.parent_organisation.website.presence || @vacancy.parent_organisation.url, class: 'govuk-link link-wrap', target: '_blank')
+            = link_to(t('schools.website_link_text', organisation_name: @vacancy.parent_organisation.name), @vacancy.parent_organisation.website.presence || @vacancy.parent_organisation.url, class: 'govuk-link link-wrap', target: '_blank')
```
Comment (david-mears-dfe) [inline app/components/jobseekers/school_overview_component.html.haml:40]: For potential git conflict future reference: in my branch I set @school to @vacancy.parent_organisation in the school overview component, to save space/duplication. (Maybe I'll attribute reader-ify it.)

File: app/models/concerns/vacancy_application_detail_validations.rb [hunk 5]
```diff
@@ -2,9 +2,11 @@ module VacancyApplicationDetailValidations
   extend ActiveSupport::Concern
 
   included do
+    validates :application_link, url: true, if: :application_link?
+
     validates :contact_email, presence: true
     validates :contact_email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i }, if: :contact_email?
 
-    validates :application_link, url: true, if: :application_link?
+    validates :contact_number, format: { with: /\A\+?(?:\d\s?){10,12}\z/ }, if: :contact_number?
```
Comment (david-mears-dfe) [inline app/models/concerns/vacancy_application_detail_validations.rb:10]: I'm curious what 10 and 12 refer to here

File: app/components/jobseekers/school_group_overview_component.html.haml (hunk 3 above)
Comment (cpjmcquillan) [reply app/components/jobseekers/school_group_overview_component.html.haml:28]: https://tools.ietf.org/html/rfc3966

File: app/models/concerns/vacancy_application_detail_validations.rb (hunk 5 above)
Comment (cpjmcquillan) [reply app/models/concerns/vacancy_application_detail_validations.rb:10]: Between 10 and 12 digits allowed

File: config/locales/en.yml [hunk 6]
```diff
@@ -189,11 +189,12 @@ en:
         supporting_documents_html:
           "Would you like to attach supporting documents? (<span class='text-red'>Required</span>)"
       application_details_form:
+        application_link: Link to online application
         contact_email_html: "Contact email (<span class='text-red'>Required</span>)"
-        school_visits: School visits
+        contact_number: Contact phone number
```
Comment (david-mears-dfe) [inline config/locales/en.yml:194]: It's missing on the Figma, but this is a required field, so `contact_number_html: "Contact phone number (<span class='text-red'>Required</span>)"`

File: app/views/hiring_staff/vacancies/application_details/show.html.haml [hunk 7]
```diff
@@ -21,6 +21,11 @@
         hint_text: t('helpers.hint.application_details_form.contact_email'),
         required: true
 
+      = f.govuk_phone_field :contact_number,
+        label: { text: t('helpers.fieldset.application_details_form.contact_number'), size: 's' },
+        hint_text: t('helpers.hint.application_details_form.contact_number'),
+        required: true
```
Comment (david-mears-dfe) [inline app/views/hiring_staff/vacancies/application_details/show.html.haml:27]: I don't think this field is required.

File: app/views/hiring_staff/vacancies/edit_vacancy_sections/_edit_application_details.html.haml [hunk 8]
```diff
@@ -14,7 +14,11 @@
     %dd.app-check-your-answers__answer.first-question
       =mail_to 'Job contact email', @vacancy.contact_email, class: 'govuk-link', 'aria-label': t('jobs.aria_labels.contact_email_link', email: @vacancy.contact_email)
 
-  - organisation = @vacancy.parent_organisation.is_a?(SchoolGroup) ? 'trust_visits' : 'school_visits'
```
Comment (david-mears-dfe) [inline app/views/hiring_staff/vacancies/edit_vacancy_sections/_edit_application_details.html.haml:17]: You don't mention removing this info in the PR. Was it accidental?

Comment (david-mears-dfe) [reply app/views/hiring_staff/vacancies/edit_vacancy_sections/_edit_application_details.html.haml:17]: Don't see a reference to this in Jira

Comment (cpjmcquillan) [reply app/views/hiring_staff/vacancies/edit_vacancy_sections/_edit_application_details.html.haml:17]: It was useless code

Comment (david-mears-dfe) [reply app/views/hiring_staff/vacancies/edit_vacancy_sections/_edit_application_details.html.haml:17]: Oh I see it's just an assignment, not a thing on the page

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**